File: Resources/lessons/Chapter 3/Examples/HiLow.java
import java.util.Random;
import java.util.Scanner;

public class HiLow {
	
	public static void main(String[] args) {
		boolean playing = true;
		Random rand = new Random();
		Scanner scan = new Scanner(System.in);
		System.out.println("Hello and welcome to the High-low number guessing game.");
		
		while (playing) {
			int answer = rand.nextInt(100) + 1;
			boolean win = false;
			int numGuesses = 1;
			
			do {
				System.out.print("Guess a number between 1 and 100:");
				int guess = scan.nextInt();
				
				if (guess == answer) {
					System.out.println("Congrats, you got in " + numGuesses + " guesses.");
					win = true;
				}
				else if (guess < answer) {
					System.out.println("Too low");
					numGuesses++;
				}
				else if (guess > answer) {
					System.out.println("Too high");
					numGuesses++;
				}
			} while (!win);
			System.out.println("Would you like to play again?(y/n)");
			String playAgain = scan.next();
			if (playAgain.equals("n")) {
				scan.close();
				playing = false;
				System.out.println("Goodbye");
			}
		}
	}
}

File: Resources/lessons/Chapter 3/Examples/HiLow.rb
playing = true

puts "Hello and welcome to the High-low number guessing game."
while (playing) do
	answer = rand(1..100)
	win = false
	numGuesses = 1
	while win == false do
		print "Guess a number between 1 and 100:"
		guess = gets.chomp.to_i
		if (guess == answer)
			print "Congrats, you got it in ", numGuesses, " guesses.\n"
			win = true
		elsif (guess < answer)
			puts "Too low"
			numGuesses += 1
		elsif (guess > answer)
			puts "Too high"
			numGuesses += 1
		end
	end
	puts "Would you like to play again?(y/n):"
	playAgain = gets.chomp
	if (playAgain == "n")
		playing = false
		puts "Goodbye"
	end
end